src/com/google/javascript/jscomp/ProductionCoverageInstrumentationCallback.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2009 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.javascript.jscomp;
+
+import static com.google.common.base.Strings.lenientFormat;
+
+import com.google.debugging.sourcemap.Base64VLQ;
+import com.google.javascript.rhino.IR;
+import com.google.javascript.rhino.Node;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+
+/**
+ * Instrument advanced coverage for javascript.
+ */
+class ProductionCoverageInstrumentationCallback extends
```

Comment:
Can this be final?

Google style is to make classes final unless they are intended to be extended